lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2078,6 +2078,12 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff)
             $sql[] = $this->getDropIndexSQL($index, $tableName);
         }
         foreach ($diff->changedIndexes as $index) {
+            if ($index->isPrimary()) {
```

Comment:
The [Scrutinizer failure](https://scrutinizer-ci.com/g/doctrine/dbal/inspections/b32d890c-c010-47f9-acf7-7aea17c498cb) reveals this as a decent candidate for refactoring in the future. Instead of the platform knowing how to drop a given index based on its properties (primary, non-primary and so on), the index should tell the platform how to drop itself (sort of a Visitor pattern):
```php
interface Index
{
    function drop(Platform $platform);
}

interface Platform
{
    function dropIndex(string $name);
    function dropPrimaryKey(string $tableName);
}
```
This is outside of scope, just noting down.

/cc @Ocramius @deeky666

Reply:
I was thinking about it but that will lead to creation of platform specific indices. Quite heavy refactoring will be required, I worry. I'm afraid I'm not able to do it in reasonable time.

Reply:
Not really fan of the suggested visitor approach. Index would then need to know how to represent itself on all existing platforms and even those not provided by DBAL out-of-box.

Reply:
@SCIF no need to worry about it right now. It's just for the record.

Reply:
> Not really fan of the suggested visitor approach. Index would then need to know how to represent itself on all existing platforms and even those not provided by DBAL out-of-box.

Not exactly. They will need to be able to identify themselves as a type while the platform will know how to handle certain types. Same as we do with column types.